dpks/quant_matrix.py

```diff
@@ -459,121 +461,205 @@ def compare_groups(
+
+            self.selector = selector
+
+        return self
+
+    def predict(
+        self,
+        classifier,
+        scaler,
+        shap_algorithm: str = "auto",
+        scale: bool = True,
+    ) -> QuantMatrix:
+
+        X = format_data(self)
+        y = encode_labels(self.quantitative_data.var["group"].values)
 
         if scale:
-            self.scaler = StandardScaler()
-            X = self.scaler.fit_transform(X)
+            X = scaler.transform(X)
 
-        self.clf = Classifier(classifier=classifier, shap_algorithm=shap_algorithm)
+        classifier = Classifier(classifier=classifier, shap_algorithm=shap_algorithm)
+
+        classifier.interpret(X)
+
+        shap_values = classifier.feature_importances_.tolist()
 
-        verbose = bool(kwargs.get("verbose", False))
-        if run_param_search:
-            param_search_method = kwargs.get("param_search_method", "genetic")
-            param_grid = dict(kwargs.get("param_grid", {}))
-            threads = int(kwargs.get("threads", 1))
-            random_state = kwargs.get("random_state", None)
-            folds = int(kwargs.get("folds", 3))
-
-            if param_search_method == "genetic":
-                gas = GeneticAlgorithmSearch(
-                    self.clf.classifier,
-                    param_grid=param_grid,
-                    threads=threads,
-                    folds=folds,
-                    n_survive=kwargs.get("n_survive", 5),
-                    pop_size=kwargs.get("pop_size", 10),
-                    n_generations=kwargs.get("n_generations", 20),
-                    verbose=verbose,
-                )
-                parameter_populations = gas.run_genetic_algorithm(X, Y)
-                self.parameter_populations = parameter_populations
-                classifier = gas.best_estimator_
-
-            elif param_search_method == "grid":
-                n_iter = int(kwargs.get("n_iter", 30))
-                scoring = str(kwargs.get("scoring", "accuracy"))
-                classifier = self.clf.get_best_estimator(
-                    X, Y, param_grid, folds, random_state, n_iter, threads, scoring
-                )
-
-        self.clf.fit(X, Y)
-
-        shap_values = self.clf.feature_importances_.tolist()
-
-        for index in drop_indexes:
-            shap_values.insert(index, np.nan)
         self.quantitative_data.obs["SHAP"] = shap_values
 
-        if calculate_feature_importance:
-            if feature_importance_method == "rfecv":
-                rfe_step = int(kwargs.get("rfe_step", 1))
-                rfe_min_features_to_select = int(
-                    kwargs.get("rfe_min_features_to_select", 1)
-                )
+        self.sample_annotations["Prediction"] = classifier.predict(X)
 
-                k_folds = int(kwargs.get("k_folds", 2))
+        return self
 
-                threads = int(kwargs.get("threads", 1))
+    def interpret(
+        self,
+        classifier,
+        scaler,
```

Comment:
Same thing with scaler here as in the predict method

Reply:
Changed, but this may lead to bugs where data for interpretation and prediction is then processed differently than during training. Might revert back